From 00db893ceabecaba57704bd3b9971db0a5429429 Mon Sep 17 00:00:00 2001
From: Roy Pledge <Roy.Pledge@freescale.com>
Date: Wed, 5 Mar 2014 14:57:45 -0500
Subject: [PATCH 144/466] Only initalize fragment owner when fragement is
 first created

When a USDPAA memory fragment is mapped into a user space process the
owner should only be initialized if the fragment has just been created.
Failure to do this will cause an error if another process is using the
fragment when the mapping is done.

Signed-off-by: Roy Pledge <Roy.Pledge@freescale.com>
Change-Id: I0fdeb195f5d1910ec70f6f56d772dd2b8cedab4c
Reviewed-on: http://git.am.freescale.net:8181/9427
Reviewed-by: Haiying Wang <Haiying.Wang@freescale.com>
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/staging/fsl_qbman/fsl_usdpaa.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_qbman/fsl_usdpaa.c b/drivers/staging/fsl_qbman/fsl_usdpaa.c
index 2de14e8..edd0126 100644
--- a/drivers/staging/fsl_qbman/fsl_usdpaa.c
+++ b/drivers/staging/fsl_qbman/fsl_usdpaa.c
@@ -979,7 +979,8 @@ do_map:
 	strncpy(start_frag->name, i->name, USDPAA_DMA_NAME_MAX);
 	start_frag->has_locking = i->has_locking;
 	init_waitqueue_head(&start_frag->wq);
-	start_frag->owner = NULL;
+	if (i->did_create == 1)
+		start_frag->owner = NULL;
 
 	/* Setup the map entry */
 	map->root_frag = start_frag;
-- 
1.7.10.4

